bowparquet.go

```diff
@@ -259,10 +257,78 @@ func (b *bow) WriteParquet(path string, verbose bool) error {
 	return nil
 }
 
-func validateColTypesMeta(b Bow, values string) (colTypesMeta []parquetColTypesMeta, err error) {
-	err = json.Unmarshal([]byte(values), &colTypesMeta)
+func (b *bow) GetParquetMetaColTimeUnit(colIndex int) (time.Duration, error) {
+	colName := b.ColumnName(colIndex)
+
+	keyIndex := b.Metadata().FindKey(keyParquetColTypesMeta)
+	if keyIndex == -1 {
+		return time.Duration(0), nil
+	}
+
+	colTypesMeta, err := readColTypesMeta(b, b.Metadata().Values()[keyIndex])
+	if err != nil {
+		return time.Duration(0), fmt.Errorf("readColTypesMeta: %w", err)
+	}
+
+	for _, m := range colTypesMeta {
+		if m.Name == colName {
+			if m.LogicalType != nil &&
+				m.LogicalType.IsSetTIMESTAMP() {
+				unit := m.LogicalType.TIMESTAMP.GetUnit()
+				if unit.IsSetMILLIS() {
+					return time.Millisecond, nil
+				} else if unit.IsSetMICROS() {
+					return time.Microsecond, nil
+				} else if unit.IsSetNANOS() {
+					return time.Nanosecond, nil
+				}
+			}
+		}
+	}
+
+	return time.Duration(0), nil
+}
+
+func NewMetaWithParquetTimestampCol(keys, values []string, colName string, timeUnit time.Duration) Metadata {
+	var colTypes = make([]parquetColTypesMeta, 1)
+
+	unit := parquet.TimeUnit{}
+	switch timeUnit {
+	case time.Millisecond:
+		unit.MILLIS = &parquet.MilliSeconds{}
+	case time.Microsecond:
+		unit.MICROS = &parquet.MicroSeconds{}
+	case time.Nanosecond:
+		unit.NANOS = &parquet.NanoSeconds{}
+	default:
+		panic(fmt.Errorf("unsupported time unit '%s'", timeUnit))
```

Comment:
Is there a standard unit ? If the unit value is not filled, do we need to panic () ?

Reply:
No standard unit, we trust the user to use a supported time unit from parquet. I added the supported time units in the function description.